app.js

```diff
@@ -14,15 +20,80 @@ if (env === 'development') {
 
 var indexRouter = require('./routes/index');
 
+
 var app = express();
 
+// passport setup
+const JwtStrategy = require('passport-jwt').Strategy;
+const ExtractJwt = require('passport-jwt').ExtractJwt;
+
+const jwtOpts = {
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+  secretOrKey: 'secret',
+  passReqToCallback: true,
+}
+
+passport.use(new JwtStrategy(jwtOpts, function(req, jwt_payload, done) {
+  const authHeader = req.header('Authorization');
+  const rawToken = authHeader ? authHeader.split(' ')[1] : '';
+
+  User.findByPk(jwt_payload.userId)
+    .then(user => {
+      console.log(req.body)
+      if (user && user.tokens.includes(rawToken)) {
+        return done(null, user);
+      } else {
+        done(null, false);
+      }
+    })
+    .catch(err => done(err, false));
+}));
+
+passport.use(
+  new Strategy(function(username, password, cb) {
+    User.findAll({ where: { username: username }, })
+      .then(user => {
+        user = user[0];
+        if (!user) { return cb(null, false); }
+
+        if (!bcrypt.compareSync(password, user.password)) {
+          return cb(null, false);
+        }
+
+        return cb(null, user);
+      })
+      .catch(err => {
+        return cb(err);
+      })
+  })
+);
+
+passport.serializeUser(function(user, cb) {
+  cb(null, user.id);
+});
+
+passport.deserializeUser(function(id, cb) {
+  User.findByPk(id)
+    .then(user  => cb(null, user))
+    .catch(err => cb(err));
+});
```

Comment:
Just a note rather than a suggestion. If the app continues to grow we could add a `./server/middlewares` folder and extract stuff like lines 26-79 out into something like `./server/middlewares/auth.js`, just to keep the main `app.js` flat.